ocs_ci/deployment/baremetal.py

```diff
@@ -320,28 +322,23 @@ def deploy(self, log_cli_level="DEBUG"):
                     self.mgmt_details[machine].get("cluster_name")
                     == constants.BM_DEFAULT_CLUSTER_NAME
                 ):
-                    secrets = [
-                        self.mgmt_details[machine]["mgmt_username"],
-                        self.mgmt_details[machine]["mgmt_password"],
-                    ]
-                    # Changes boot prioriy to pxe
-                    cmd = (
-                        f"ipmitool -I lanplus -U {self.mgmt_details[machine]['mgmt_username']} "
-                        f"-P {self.mgmt_details[machine]['mgmt_password']} "
-                        f"-H {self.mgmt_details[machine]['mgmt_console']} chassis bootdev pxe"
-                    )
-                    run_cmd(cmd=cmd, secrets=secrets)
-                    sleep(2)
-                    # Power On Machine
-                    cmd = (
-                        f"ipmitool -I lanplus -U {self.mgmt_details[machine]['mgmt_username']} "
-                        f"-P {self.mgmt_details[machine]['mgmt_password']} "
-                        f"-H {self.mgmt_details[machine]['mgmt_console']} chassis power cycle || "
-                        f"ipmitool -I lanplus -U {self.mgmt_details[machine]['mgmt_username']} "
-                        f"-P {self.mgmt_details[machine]['mgmt_password']} "
-                        f"-H {self.mgmt_details[machine]['mgmt_console']} chassis power on"
-                    )
-                    run_cmd(cmd=cmd, secrets=secrets)
+                    if self.mgmt_details[machine]["role"] == "bootstrap":
+                        self.set_pxe_boot_and_reboot(machine)
+
+                    elif (
+                        self.mgmt_details[machine]["role"] == constants.MASTER_MACHINE
+                        and master_count <= config.ENV_DATA["master_replicas"]
```

Comment:
Here I think it should be `and master_count < config.ENV_DATA["master_replicas"]`

Reply:
Done